lib/resque/worker.rb

```diff
@@ -610,7 +610,14 @@ def prune_dead_workers
         # client library or an older version of Resque. We won't touch these.
         if all_workers_with_expired_heartbeats.include?(worker)
           log_with_severity :info, "Pruning dead worker: #{worker}"
-          worker.unregister_worker(PruneDeadWorkerDirtyExit.new(worker.to_s))
+
+          job_class = begin
+            worker.job(false)['payload']['class']
+          rescue => _error
+            "<Unknown Job>"
```

Comment:
Since `PruneDeadWorkerDirtyExit` is responsible for creating the exception message, would set `job_class` to `nil` here and let `PruneDeadWorkerDirtyExit#initialize` use e.g. `job ||= "<Unknown Job>"`